src/Dapr.Extensions.Configuration/DaprConfigurationStoreSource.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using Dapr.Client;
+using Microsoft.Extensions.Configuration;
+
+namespace Dapr.Extensions.Configuration
+{
+    /// <summary>
+    /// Configuration source that provides a <see cref="DaprConfigurationStoreProvider"/>.
+    /// </summary>
+    [Obsolete]
+    public class DaprConfigurationStoreSource : IConfigurationSource
+    {
+        /// <summary>
+        /// The Configuration Store to query.
+        /// </summary>
+        public string Store { get; set; } = default!;
+
+        /// <summary>
+        /// The list of keys to request from the configuration. If empty, request all keys.
+        /// </summary>
+        public IReadOnlyList<string> Keys { get; set; } = default!;
+
+        /// <summary>
+        /// The <see cref="DaprClient"/> used to query the configuration.
+        /// </summary>
+        public DaprClient Client { get; set; } = default!;
+
+        /// <summary>
+        /// Boolean stating if this is a streaming call or not.
+        /// </summary>
+        public bool IsStreaming { get; set; } = false;
+
+        /// <summary>
+        /// Gets or sets the <see cref="TimeSpan"/> that is used to control the timeout waiting for the Dapr sidecar to become healthly.
+        /// </summary>
+        public TimeSpan SidecarWaitTimeout { get; set; }
+
+        /// <summary>
+        /// The optional metadata to be sent to the configuration store.
+        /// </summary>
+        public IReadOnlyDictionary<string, string>? Metadata { get; set; } = default;
```

Comment:
Does the null-ness of this field affect the semantics/behavior? If not, then it would be better to initialize this to a non-null value.

Reply:
I don't believe it does. I'll try it with an empty dictionary and see if things still work as expected.